src/components/ui/date-field.tsx

```diff
@@ -85,7 +85,7 @@ export default function DateField({
   };
 
   const handleYearChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const newYear = e.target.value;
+    const newYear = e.target.value.slice(0, 4);
```

Comment:
this makes UX bad if they move their caret to left and start typing. numbers will start disappearing and carets start jumping here and there.